src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,59 @@
+package duke.parser;
+
+import duke.command.*;
+import duke.exception.DukeException;
+import duke.exception.InvalidCommandException;
+
+/**
+ * Represents the parser for program to parse input.
+ */
+public class Parser {
+    // Parser to return the action to do maybe the command to do
+    // input : String
+    // from that input we will split the words into command / description / datetime
+    private String userInputLine;
+
+    /**
+     * Creates new Parser.
+     *
+     * @param userInputLine UserInputLine given for parser to parse.
+     */
+    public Parser(String userInputLine) {
+        this.userInputLine = userInputLine;
+    }
+
+    /**
+     * Returns Command that user wants to execute.
+     *
+     * @return Command to be executed on a TaskList.
+     * @throws DukeException If userInputLine invalid for the command.
+     */
+    public Command parse() throws DukeException {
+        // split command away
+        String[] inputStrings = this.userInputLine.trim().split(" ", 2);
+        String inputCommand = inputStrings[0];
+        // has command and body
+        String inputBody = inputStrings.length == 2 ? inputStrings[1] : "";
+        switch (inputCommand) {
+        case "list":
+            return new ListCommand("");
```

Comment:
more nit-picky then anything, I find it a little odd a new object is initialised with an empty string  but don't worry, I understand that the `Command` parent class required you to give a parameter 😆